core-errors/src/lib.rs

```diff
@@ -26,277 +26,354 @@ extern crate alloc;
 mod simple;
 
 use core::fmt::Debug;
+use enum_iterator::Sequence;
 #[cfg(feature = "codec")]
-use scale_info::{
-    scale::{self, Decode, Encode},
-    TypeInfo,
+use {
+    alloc::vec::Vec,
+    scale_info::{
+        scale::{Decode, Encode, Error, Input},
+        TypeInfo,
+    },
 };
 
 pub use simple::*;
 
-/// Error using messages.
-#[allow(clippy::unnecessary_cast)]
-#[derive(Copy, Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd, derive_more::Display)]
-#[cfg_attr(feature = "codec", derive(Encode, Decode, TypeInfo), codec(crate = scale))]
+/// Execution error.
+#[derive(
+    Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Sequence, derive_more::Display,
+)]
 #[non_exhaustive]
-#[repr(u8)]
-pub enum MessageError {
-    /// Message has bigger then allowed one message size
-    #[display(fmt = "Max message size exceed")]
-    MaxMessageSizeExceed = 0,
+#[repr(u32)]
+pub enum ExecutionError {
+    /// An error occurs in attempt to wait duration greater than could be paid.
```

Comment:
Is it right comment and info string? I can see this error may occur in many another cases